src/services/fileDestinationUrl.service.ts

```diff
@@ -0,0 +1,49 @@
+import { S3 } from 'aws-sdk';
+import { v4 as uuidv4 } from 'uuid';
+
+const createFileDestinationUrl = async ({
+  bucket,
+  fileType,
+  maxSize = 0,
+  fileName = '',
+  path = '',
+}: {
+  bucket: string;
+  fileType: string;
+  maxSize?: number;
+  fileName?: string;
+  path?: string;
```

Comment:
Ah -- when we spoke on that I was scoped to just the single parameter under discussion (fileType).  I didn't realize we were talking about NO optional parameters.

I'm OK with `maxSize` being required for sure.

I believe that `path` makes more sense as optional with a default of `/` -- I'm not sure what we gain from requiring `/` to be passed.

`fileName` being required feels like the incorrect API design (file names can have collisions in a bucket).  Allowing fileName allows a client to explicitly write over an existing file, but requiring fileName makes it possible for a client to *accidentally* write over an existing file.

NOTE: Since I moved to POST requests aws might not actually allow data to overwrite (gonna test that now), but even still this would make the calling client responsible for avoiding collisions despite having no way of discovering what is / is not a collision.

Reply:
Ah sorry, there are two parts here which I conflated a bit!

1. Should the TypeScript type definition for this method allow for optional parameters? This is what I meant to focus on with this review comment! Since we're providing default values in the method, the *type* of these arguments cannot ever be `undefined`: they're either the value the caller passed in, or the default value provided. In this narrow technical context, I think it's safe and uncontroversial to say that these should all be non-optional.
1. Should the API require these parameters? This is the broader question, and I'm glad you noted it, since I didn't realize we weren't completely on the same page! I'm generally of the opinion that fewer parameters - and especially, fewer optional parameters - is better. I think we could drop `fileName` and only use UUIDs; particularly if upon further research you discover that you cannot overwrite via POST. Dropping `path` is less clear, particularly since we need to support some kind of path prefix for the S3 bucket for the local dev environments; that could be optional, although I'm also slightly tempted to argue for an environment variable. I haven't yet thought through what the local development process is going to look like for integrating this service.

Reply:
The environment variable question is a great one to raise; I actually was on the fence about it for bucket being defined by environment as well.

I'll finish the first pass updating based on this feedback / thought process and we can continue the discussion from there.

Reply:
for future reference: POST *does* allow you to overwrite files.